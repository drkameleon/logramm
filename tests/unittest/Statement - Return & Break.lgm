
func(x) =>
{
	out "func(x)--> about to return...";
	==> x*2;
	out "func(x)--> should have returned... :-(";
}

funcLoop(x) =>
{
	items = [1..10];

	@ i -> items : {
		i == 5 : { 
			out "Found '5'. Returning! "; 
			==> i;
		}
		out "Looping. I = " + i;
	}

	out "After loop.";
	==> 100;
}

out "Let's see how Return works";
out "I mean, other than pushing a return value to the stack...";
out "Does it really return?";

out "Calculation f(2) = 2*2 = " + func(2);
out "======================";

out "Let's see if that works with a loop...";
out funcLoop(10);

out "HERE";

@ i -> [1..10] :
{
	out "i = {:i}";
	i==5: { out "breaking..."; break; }
	out "After i = {:i}";
}

out "OUT of break";

funcRetLoopBreak() =>
{
	out "in funcRetLoopBreak";
	@ i -> [1..10]:
	{
		out "i = " + i;
		@ x -> [1..10]:
		{
			out "x = " + x;
			i==3 and x == 6 : break;
		}
		out "out of loop";
		i==3 and x == 6 : ==> "done";
	}
}

out funcRetLoopBreak();

@ k -> [1..10] : {}
out "k = " + k;

@ k -> [1..10] : k==3: break;
out "k = " + k;
