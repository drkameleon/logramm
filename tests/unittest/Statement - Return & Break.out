Let's see how Return works
I mean, other than pushing a return value to the stack...
Does it really return?
func(x)--> about to return...
Calculation f(2) = 2*2 = 4
======================
Let's see if that works with a loop...
Looping. I = 1
Looping. I = 2
Looping. I = 3
Looping. I = 4
Found '5'. Returning! 
5
HERE
i = 1
After i = 1
i = 2
After i = 2
i = 3
After i = 3
i = 4
After i = 4
i = 5
breaking...
OUT of break
in funcRetLoopBreak
i = 1
x = 1
x = 2
x = 3
x = 4
x = 5
x = 6
x = 7
x = 8
x = 9
out of loop
i = 2
x = 1
x = 2
x = 3
x = 4
x = 5
x = 6
x = 7
x = 8
x = 9
out of loop
i = 3
x = 1
x = 2
x = 3
x = 4
x = 5
x = 6
out of loop
done
k = 9
k = 3
